shell/src/main/java/org/apache/accumulo/shell/commands/ScanCommand.java

```diff
@@ -285,6 +288,19 @@ protected void fetchColumns(final CommandLine cl, final ScannerBase scanner,
     }
   }
 
+  protected String[] extractColumnFamily(final String columnString) {
```

Comment:
This might be able to be simplified by just using the substring method. Also it doesn't seem like this accounts for strings where there are more than 2 ':' characters (not sure if that will ever be the case), for example `a:c:d:e` would be sent to the `else` branch and possibly handled incorrectly (not sure if that will ever be the case).

Reply:
Are referring to that we just take a substring from the last index of : and that would do it?  If we have more than 2 that will be a problem, I can change the condition because anything  &gt; 2 should be split into two strings cf and qualifier.

Reply:
The colFam and the colQual values are user defined so Accumulo really should be able to handle whatever they wish to do.  Unless there are specific characters that are prohibited, then they should be allowed.

One example could be an ipv6 address say colFam was IP and colQual was port - you may want to scan for a specific IP and specific port, just the IP, or even maybe IP that starts with and either with or without a port.

Using the IPv6 example from [wikipedia](https://en.wikipedia.org/wiki/IPv6_address) ```2001:0db8:85a3:0000:0000:8a2e:0370:7334``` If that was a colFam, then it would be desirable to scan for all or parts of that address plus (optionally) any colQual value that a user might want.

Reply:
makes sense, we should be able to fix it easily

Reply:
If that is the case, then having a separate opt for cf and cq could be easier. Take the entire string after the -cf opt instead of trying to parse everything accordingly.

Reply:
If we are taking input in this form 2001:0db8:85a3:0000:0000:8a2e:0370:7334 its hard to know what is part of cf and what is sq. I think if we would be seeing this then we should move to using options  to specify the column family and column qualifier using cf and cq.

Reply:
> Are referring to that we just take a substring from the last index of : and that would do it?

Yea what I'm saying is, the substring method you have will correctly handle all cases so you could remove the else branch. @harjitdotsingh

Reply:
so if we go the cf and cq route do we get rid of -c or keep it ?

Reply:
I'd vote to get rid of it since cf and cq covers its functionality.

Reply:
Here are the comments from @ctubbsii  

`Alternatively, we can add -cf and -cq options to the scan command which is mutually exclusive with -c.`

Going by that  it would mean that we still want to keep -c but  IMHO its redundant to have  -c , -cf and -cq.  Would like to see what @ctubbsii  has to say about it? Do we want to keep -c with -cf and -cq. I think it will be confusing for people to have 2 options for a operation. Thoughts ?

Reply:
It is likely that people have scripts that use -c. Removing that as an option will break those scripts.  People using -c currently are limited to not having : in the colFam, but they are coping.

Adding -cq and -cf provide additional capability when (the current -c) is insufficient for their needs.